disco/state.py

```diff
@@ -285,6 +290,11 @@ def on_guild_member_add(self, event):
         if event.member.guild_id not in self.guilds:
             return
 
+        if (self.guilds[event.member.guild_id].member_count is not UNSET and
```

Comment:
I'm not sure I love this behavior being built into the lib, but I'd have to think on it more. For now I'd move this to a separate PR.